File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {

    List<List<Integer>> result = new ArrayList<>();
    List<Integer> currentResult = new ArrayList<>();
    
    public void dfs(int[] nums, int index, int target, List<Integer> currResult) {

        if (target == 0) {
            result.add(new ArrayList<>(currResult));
            currResult = new ArrayList<>();
            return;
        }

        if (target < 0) {
            return;
        }

        for (int i = index; i < nums.length; i++) {
            currResult.add(nums[i]);
            dfs(nums, i, target - nums[i], currResult);
            currResult.remove(currResult.size() - 1);
        }
    }

    public List<List<Integer>> combinationSum(int[] nums, int target) {

        for (int i = 0; i < nums.length; i++) {
            currentResult.add(nums[i]);
            dfs(nums, i, target - nums[i], currentResult);
            currentResult.remove(currentResult.size() - 1);
        }
        return result;
    }
}

File: Data Structures & Algorithms/count-paths/submission-0.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        arr = [[False for _ in range(m)] for _ in range(n)]
        def dfs(i, j):
            if i < 0 or j < 0:
                return 0

            if i == 0 and j == 0:
                return 1 

            return dfs(i - 1, j) + dfs(i, j - 1)
        
        return dfs(m - 1, n - 1)